### PR TITLE
fix regression in join vertices post process.

### PR DESCRIPTION
**Pull request by ockeymm123**
If a bone has too many 'mWeights'(eg : bone->mNumWeights =  50000+), It may takes more than half a minute to adjust bone vertex weights.
We can use mark '0x80000000' to remove the these duplicating weights.
reference commit id: a22aa75bca5da546abb320edefe2da0ae3316535

### CODE REVIEW

File: code/PostProcessing/JoinVerticesProcess.cpp [hunk 1]
```diff
@@ -400,17 +401,8 @@ int JoinVerticesProcess::ProcessMesh( aiMesh* pMesh, unsigned int meshIndex) {
             for ( unsigned int b = 0; b < bone->mNumWeights; b++ ) {
                 const aiVertexWeight& ow = bone->mWeights[ b ];
                 // if the vertex is a unique one, translate it
+				// filter out joined vertices by mrak.
```
Comment (turol) [inline code/PostProcessing/JoinVerticesProcess.cpp:404]: Typo

**Review by kimkulling — APPROVED** [review]
Looks fine.

File: code/PostProcessing/JoinVerticesProcess.cpp [hunk 2]
```diff
@@ -357,7 +357,8 @@ int JoinVerticesProcess::ProcessMesh( aiMesh* pMesh, unsigned int meshIndex) {
             }
         } else{
             // if the vertex is already there just find the replace index that is appropriate to it
-            replaceIndex[a] = it->second;
+			// mark it with '0x80000000'
+            replaceIndex[a] = it->second | 0x80000000;
```
[the remark this answers is not in the document]
Comment (kimkulling) [reply code/PostProcessing/JoinVerticesProcess.cpp]: Fixed.

**Comment by kimkulling** [comment]
Merged, thanks a lot for your contribution.

**Pull request merged**